kde/ki18n/Portfile

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+# Based on Portfile for kde-cmake-extra-modules
+
+# Potentially to be replaced as part of
+# https://trac.macports.org/ticket/49925
+
+PortSystem          1.0
+PortGroup           qt5     1.0
+PortGroup           cmake   1.1
+
+name                ki18n
+subport ${name}-devel {}
+
+set kf5.branch      5.47
+
+if {${subport} eq "${name}-devel"} {
+    fetch.type      git
+    git.url         https://anongit.kde.org/ki18n.git
+#    v5.47.0
+    git.branch      5faba1bbc39588c120e22d7cf619322b441cd49f
+    version         5.47.0
+} else {
+    version         ${kf5.branch}.0
+    set branch      [join [lrange [split ${version} .] 0 1] .]
+    master_sites    https://download.kde.org/stable/frameworks/${branch}/
+
+    distname        ${name}-${version}
+    use_xz          yes
+
+    checksums       rmd160  5525340b79b1d8f477258adad392ad680a0d7358 \
+                    sha256  2627faa0186e9b463d978e842bf6df1e7d27e7e9d487793f6709e82a1c0729a8 \
+                    size    586888
+}
+
+categories          kde kf5 devel
+license             GPL-2+
+maintainers         {@chrstphrchvz gmx.us:chrischavez} openmaintainer
+description         KDE Gettext-based UI text internationalization
+long_description    \
+"KI18n provides functionality for internationalizing user interface text
+in applications, based on the GNU Gettext translation system.
+It wraps the standard Gettext functionality, so that the programmers
+and translators can use the familiar Gettext tools and workflows.
+
+KI18n provides additional functionality as well, for both programmers
+and translators, which can help to achieve a higher overall quality
+of source and translated text. This includes argument capturing,
+customizable markup, and translation scripting."
+
+platforms           darwin
+homepage            https://projects.kde.org/projects/kdesupport/ki18n
+
+qt5.depends_component \
```

Comment:
KI18n v5.42.0 also depended on QtDeclarative. Please double-check; a priori QtScript is obsolete and code should migrate to QML so it seems unlikely that the dependency has been dropped.

Reply:
Looks like QtDeclarative/QML is used in tests only - https://github.com/KDE/ki18n/search?utf8=%E2%9C%93&q=qml&type=, and there's currently no `test` defined in this port. (Of course adding test definitions would be nice) On the other hand, QtScript is recommended by KDE developers:
```
option(BUILD_WITH_QTSCRIPT "Build with support for scripted translations (recommended)" ON)
```
(https://github.com/KDE/ki18n/blob/f935c698b5ad1fb55fbd093ea1719d7a38f7f564/CMakeLists.txt#L34)

Reply:
The dependencies I put were naïvely based on checking what other package managers use/getting through the configure phase successfully (rather than knowing what's in ki18n), which is where I saw QtScript but no mention of QtDeclarative.
<span><details><summary>Configure phase output:</summary>
<pre>
--->  Configuring kf5-ki18n
        (using ccache)
Executing:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports_macports-ports_kde_kf5-ki18n/kf5-ki18n/work/build" && /opt/local/bin/cmake -G "CodeBlocks - Unix Makefiles" -DCMAKE_BUILD_TYPE=MacPorts -DCMAKE_INSTALL_PREFIX="/opt/local" -DCMAKE_INSTALL_NAME_DIR="/opt/local/lib" -DCMAKE_SYSTEM_PREFIX_PATH="/opt/local;/usr" -DCMAKE_C_COMPILER_LAUNCHER=/opt/local/bin/ccache -DCMAKE_CXX_COMPILER_LAUNCHER=/opt/local/bin/ccache -DCMAKE_C_COMPILER="$CC" -DCMAKE_CXX_COMPILER="$CXX" -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_POLICY_DEFAULT_CMP0060=NEW -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_COLOR_MAKEFILE=ON -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_MODULE_PATH="/opt/local/share/cmake/Modules" -DCMAKE_PREFIX_PATH="/opt/local/share/cmake/Modules" -DCMAKE_BUILD_WITH_INSTALL_RPATH:BOOL=ON -DCMAKE_INSTALL_RPATH="/opt/local/lib" -Wno-dev -DECM_MKSPECS_INSTALL_DIR=/opt/local/libexec/qt5/mkspecs/modules -DCMAKE_OSX_ARCHITECTURES="x86_64" -DCMAKE_OSX_DEPLOYMENT_TARGET="10.13" -DCMAKE_OSX_SYSROOT="/" /opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports_macports-ports_kde_kf5-ki18n/kf5-ki18n/work/ki18n-5.47.0 
-- The C compiler identification is AppleClang 9.1.0.9020039
-- The CXX compiler identification is AppleClang 9.1.0.9020039
-- Check for working C compiler: /usr/bin/clang
-- Check for working C compiler: /usr/bin/clang -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/clang++
-- Check for working CXX compiler: /usr/bin/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- 

-- Looking for __GLIBC__
-- Looking for __GLIBC__ - not found
-- Performing Test _OFFT_IS_64BIT
-- Performing Test _OFFT_IS_64BIT - Success
-- Performing Test HAVE_DATE_TIME
-- Performing Test HAVE_DATE_TIME - Success
-- Could not set up the appstream test. appstreamcli is missing.
CMake Warning at /opt/local/share/ECM/kde-modules/KDECMakeSettings.cmake:270 (message):
  Searching for X11 and related technologies is disabled on Apple systems.
  Set APPLE_FORCE_X11 to ON to change this behaviour.  Set
  APPLE_SUPPRESS_X11_WARNING to ON to hide this warning.
Call Stack (most recent call first):
  CMakeLists.txt:41 (include)


-- Looking for dngettext
-- Looking for dngettext - not found
-- libintl is a separate library.
-- Found LibIntl: /opt/local/include  
-- Found PythonInterp: /usr/bin/python (found version "2.7.10") 
-- Found Gettext: /opt/local/bin/msgmerge (found version "0.19.8.1") 
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
-- The sip executable must be available to use PythonModuleGeneration.
-- The following REQUIRED packages have been found:

 * ECM (required version >= 5.47.0), Extra CMake Modules., <https://projects.kde.org/projects/kdesupport/extra-cmake-modules>
 * Qt5Script (required version >= 5.8.0)
 * Qt5Core
 * LibIntl, <http://gnuwin32.sourceforge.net/packages/libintl.htm>
   Needed for building KI18n unless glibc is the system libc implementation
 * PythonInterp
 * Gettext
 * Qt5Test
 * Qt5Network (required version >= 5.10.1)
 * Qt5Qml
 * Qt5Concurrent
 * Qt5 (required version >= 5.8.0)

-- The following features have been disabled:

 * QCH, API documentation in QCH format (for e.g. Qt Assistant, Qt Creator & KDevelop)

-- The following OPTIONAL packages have not been found:

 * PythonModuleGeneration

-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_POLICY_DEFAULT_CMP0025


-- Build files have been written to: /opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports_macports-ports_kde_kf5-ki18n/kf5-ki18n/work/build
</pre>
</details><br /></span>

If I add `test.run yes` to the Portfile I can get these test results:
```
--->  Testing kf5-ki18n
Executing:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports_macports-ports_kde_kf5-ki18n/kf5-ki18n/work/build"&& /usr/bin/make test
Running tests...
/opt/local/bin/ctest --force-new-ctest-process
Test project /opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports_macports-ports_kde_kf5-ki18n/kf5-ki18n/work/build
    Start 1: ki18n-declarativetest
1/5 Test #1: ki18n-declarativetest ............   Passed    0.10 sec
    Start 2: ki18n-klocalizedstringtest
2/5 Test #2: ki18n-klocalizedstringtest .......   Passed    0.08 sec
    Start 3: ki18n-ktranscripttest
3/5 Test #3: ki18n-ktranscripttest ............***Failed    0.06 sec
    Start 4: ki18n-ktranscriptcleantest
4/5 Test #4: ki18n-ktranscriptcleantest .......   Passed    0.06 sec
    Start 5: ki18n_install
5/5 Test #5: ki18n_install ....................   Passed    5.55 sec

80% tests passed, 1 tests failed out of 5

Total Test time (real) =   5.86 sec

The following tests FAILED:
          3 - ki18n-ktranscripttest (Failed)
Errors while running CTest
make: *** [test] Error 8
Command failed:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports_macports-ports_kde_kf5-ki18n/kf5-ki18n/work/build" && /usr/bin/make test
Exit code: 2
```